pkg/build/registry/buildconfig/rest.go

```diff
@@ -75,6 +77,11 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 	if errs := validation.ValidateBuildConfig(buildConfig); len(errs) > 0 {
 		return nil, errors.NewInvalid("buildConfig", buildConfig.Name, errs)
 	}
+
+	if circularDeps(buildConfig) {
```

Comment:
And by that I mean we have to run an O(N) load on the cluster for every create, which is extremely expensive.  Ben and I talked through a number of options, but none of them are really efficient or easy to implement, and they don't solve the problem for end users.

At a minimum, `osc status` will show some of this for end users and that will reduce the possibility this happens.